AmplifyPlugins/API/AWSAPICategoryPlugin/Support/Internal/URLComponents+sigv4Encoding.swift

```diff
@@ -41,14 +42,15 @@ extension URLComponents {
         percentEncodedQuery = try queryItems.map { name, value in
             guard let encodedName = encodeQueryParamItemBySigV4Rules(name),
                   let encodedValue = encodeQueryParamItemBySigV4Rules(value) else {
-                throw SigV4Error.invalidQueryItem(name, value)
+                throw APIError.invalidURL(
+                    "Invalid query parameter.",
+                    """
+                    Review your API plugin configuration and make sure to pass valid query parameters in your request.
```

Comment:
Nit: Would the plugin configuration be the issue, or is it more likely to be something in app code or even user-provided data? Maybe something like:
```suggestion
                    Review your Amplify.API call to make sure you are passing valid UTF-8 query parameters in your request.
```

(Appropriately line-breaked, of course :) )